src/main/java/at/hannibal2/skyhanni/utils/tracker/SkyHanniBucketedItemTracker.kt

```
@@ -0,0 +1,214 @@
+    drawDisplay: (BucketedData) -> List<Searchable>,
+) : SkyHanniTracker<BucketedData>(name, createNewSession, getStorage, drawDisplay) {
+
+    companion object {
+        val SKYBLOCK_COIN = NEUInternalName.SKYBLOCK_COIN
+    }
+
+    fun addCoins(bucket: E, coins: Int) {
+        addItem(bucket, SKYBLOCK_COIN, coins)
+    }
+
+    fun addItem(bucket: E, internalName: NEUInternalName, amount: Int) {
+        modify {
+            it.addItem(bucket, internalName, amount)
+        }
+        getSharedTracker()?.let {
+            val hidden = it.get(DisplayMode.TOTAL).getItemsProp()[internalName]!!.hidden
+            it.get(DisplayMode.SESSION).getItemsProp()[internalName]!!.hidden = hidden
+        }
+
+        val (itemName, price) = SlayerAPI.getItemNameAndPrice(internalName, amount)
+        if (config.warnings.chat && price >= config.warnings.minimumChat) {
+            ChatUtils.chat("§a+Tracker Drop§7: §r$itemName")
+        }
+        if (config.warnings.title && price >= config.warnings.minimumTitle) {
+            LorenzUtils.sendTitle("§a+ $itemName", 5.seconds)
+        }
+    }
+
+    fun addPriceFromButton(lists: MutableList<Searchable>) {
+        if (isInventoryOpen()) {
+            lists.addSearchableSelector<ItemPriceSource>(
+                "",
+                getName = { type -> type.sellName },
+                isCurrent = { it.ordinal == config.priceSource.ordinal }, // todo avoid ordinal
+                onChange = {
+                    config.priceSource = ItemPriceSource.entries[it.ordinal] // todo avoid ordinal
+                    update()
+                },
+            )
+        }
+    }
+
+    inline fun <reified F : Enum<F>> addBucketSelectorsReified(
+        lists: MutableList<Searchable>,
+        data: BucketedData,
+        sourceString: String = "",
+    ) {
+        if (isInventoryOpen()) {
+            val sourceStringPrefix = sourceString.takeIf { it.isNotEmpty() } ?: "Item Source"
```

have sourceStringPrefix as a constructor parameter to the class instead. also rather throw an error in dev env when the string is empty isntead of showing a generic name